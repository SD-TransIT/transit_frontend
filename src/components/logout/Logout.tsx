import React from 'react';

import { useNavigate } from 'react-router-dom';

import { Paths } from 'routes/paths';
import vector from 'shared/images/vector.svg';
import { sessionToken } from 'stores/reducers/tokenReducer';

function Logout() {
  const navigate = useNavigate();

  const callback = () => {
    navigate(Paths.sign_in);
  };

  const logoutUser = () => {
    localStorage.removeItem(sessionToken);
    callback();
  };

  return (
    <div className="flex justify-end h-full items-center pr-2" data-testid="logout">
      <button type="button" onClick={() => logoutUser()} className="flex flex-row gap-3">
        <img src={vector} alt="logout" className="m-auto" />
        <p>Logout</p>
      </button>
    </div>
  );
}

export default Logout;
