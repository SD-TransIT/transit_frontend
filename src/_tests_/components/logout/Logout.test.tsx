import React from 'react';

import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Logout from 'components/logout/Logout';

describe('Unit test for the Logout component.', () => {
  const LOGOUT_TEST_ID = 'logout';

  test('Should match snapshot.', () => {
    render(
      <MemoryRouter initialEntries={[{ pathname: '/', search: '/' }]}>
        <Logout />
      </MemoryRouter>,
    );
    expect(screen.queryByTestId(LOGOUT_TEST_ID)).toMatchSnapshot();
  });
});
