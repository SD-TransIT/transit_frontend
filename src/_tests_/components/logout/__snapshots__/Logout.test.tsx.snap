// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the Logout component. Should match snapshot. 1`] = `
<div
  class="flex justify-end h-full items-center pr-2"
  data-testid="logout"
>
  <button
    class="flex flex-row gap-3"
    type="button"
  >
    <img
      alt="logout"
      class="m-auto"
      src="vector.svg"
    />
    <p>
      Logout
    </p>
  </button>
</div>
`;
